src/main/CommandHandler.cpp

```diff
@@ -725,17 +710,8 @@ CommandHandler::upgrades(std::string const& params, std::string& retStr)
         auto addParam = [&](std::string const& name,
                             stellar::optional<uint32>& f) {
             uint32 v;
-            bool updated;
-            if (!parseNumParam(retMap, name, v, retStr,
-                               Requirement::OPTIONAL_REQ, updated))
-            {
-                retStr = (fmt::MemoryWriter()
-                          << fmt::format("could not parse {}: '{}'\n", name,
-                                         retMap[name])
-                          << retStr)
-                             .str();
-            }
-            else if (updated)
+            auto res = parseNumParam(retMap, name, v);
+            if (res)
```

Comment:
isn't that code equivalent to `f = parseNumParam(...)` ?
If so, this lambda is now obsolete

Reply:
I think you're right. `reset()` would result in the same behavior as assigning `nullopt` (which is `shared_ptr`() under the hood)